engine/chain/internal/combined_txs_getter.go

```diff
@@ -0,0 +1,71 @@
+package internal
+
+import (
+	"context"
+	"maps"
+
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/errors"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/models"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"iter"
+)
+
+// CombinedTxsGetter is a TransactionsGetter that combines multiple TransactionsGetters
+type CombinedTxsGetter struct {
```

Comment:
Could it be private?

Reply:
it is private now. 
But I've also reorganized things here; partially as you suggested to use `"Combine"` word  xD - check it out 👀 👍